src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -107,16 +108,31 @@ TxQ::FeeMetrics::update(Application& app,
         // so clamp down on limits.
         txnsExpected_ = boost::algorithm::clamp(feeLevels.size(),
             minimumTxnCount_, targetTxnCount_);
+        recentTxnCounts_.clear();
     }
     else if (feeLevels.size() > txnsExpected_ ||
         feeLevels.size() > targetTxnCount_)
     {
+        recentTxnCounts_.emplace_back(feeLevels.size());
+        while (recentTxnCounts_.size() > setup.ledgersInQueue)
```

Comment:
This bit of code suggests that a `std::deque` may not be the right choice here. Have we considered [`boost::circular_buffer`](http://www.boost.org/doc/libs/1_65_1/doc/html/circular_buffer.html)?

Reply:
I had not, but that's a great idea.